test/unit/host_test.rb

```diff
@@ -674,16 +674,42 @@ def teardown
         :ptable => FactoryGirl.create(:ptable, :operatingsystem_ids => [operatingsystems(:redhat).id]), :medium => media(:one),
         :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :subnet => subnets(:two), :puppet_proxy => smart_proxies(:puppetmaster),
         :architecture => architectures(:x86_64), :environment => environments(:production), :managed => true,
-        :owner_type => "User", :root_pass => "xybxa6JUkz63w"
-      assert host.valid?
+        :owner_type => "User", :owner => User.current, :root_pass => "xybxa6JUkz63w"
+      assert_valid host
     end
 
     test "should not save if owner_type is not User or Usergroup" do
       host = Host.new :name => "myfullhost", :mac => "aabbecddeeff", :ip => "3.3.4.03", :medium => media(:one),
         :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :subnet => subnets(:two), :puppet_proxy => smart_proxies(:puppetmaster),
         :architecture => architectures(:x86_64), :environment => environments(:production), :managed => true,
         :owner_type => "UserGr(up" # should be Usergroup
-      assert !host.valid?
+      refute_valid host
+    end
+
+    test 'should succeed validation if owner not set' do
```

Comment:
nitpick, these tests are good, but they could use factories & be put in a context to be much easier to read - https://gist.github.com/ad4e32e5f751f956918362f79c6851ea